### ISSUE TITLE
FFmpegFrameGrabber.start() crashes when loading mp4 file

### ISSUE THREAD

**Issue by parejadan** [bytedeco/javacv#640]
I'm trying to load a mp4 video file I record somewhere else in the app, and then want to pass the image's address to a jni native opencv code to do some processing. 

long story short it crashes on frameGrabber.start();

literally switched from opencv 30 minutes ago and hacked it all together at last minute...-__-

```
        String videoPath = videoPaths.get(videoPaths.size()-1);

        FrameGrabber frameGrabber = new FFmpegFrameGrabber(videoPath);

        try {
            frameGrabber.start();
            Frame fImg = frameGrabber.grab();
            OpenCVFrameConverter.ToMat converterToMat = new OpenCVFrameConverter.ToMat();
            opencv_core.Mat mImg = converterToMat.convert(fImg);

            Log.e(TAG, String.format("width: %d\t|height: %d\n", mImg.cols(), mImg.rows()));
            _lbs.HogDetect(mImg.address());
            String fileNam = getProcImgPath(getApplicationContext());
            opencv_core.cvSave(fileNam, mImg.ptr());

            displayFeedback(fileNam);

        } catch(Exception e) {
            Log.d(TAG, "can't open vid");
            System.exit(-1);
        }

```
> 03-11 17:13:16.817 11910-11910/firenation.beepbopP.ember E/AndroidRuntime: FATAL EXCEPTION: main
>                                                                            Process: firenation.beepbopP.ember, PID: 11910
>                                                                            java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/firenation.beepbopP.ember-1/base.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_dependencies_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_0_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_1_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_2_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_3_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_4_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_5_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_6_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_7_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_8_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_9_apk.apk"],nativeLibraryDirectories=[/data/app/firenation.beepbopP.ember-1/lib/arm64, /data/app/firenation.beepbopP.ember-1/base.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_dependencies_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_0_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_1_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_2_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_3_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_4_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_5_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_6_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_7_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_8_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_9_apk.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]] couldn't find "libjniavutil.so"
>                                                                                at java.lang.Runtime.loadLibrary0(Runtime.java:972)
>                                                                                at java.lang.System.loadLibrary(System.java:1530)
>                                                                                at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:804)
>                                                                                at org.bytedeco.javacpp.Loader.load(Loader.java:613)
>                                                                                at org.bytedeco.javacpp.Loader.load(Loader.java:530)
>                                                                                at org.bytedeco.javacpp.avutil.<clinit>(avutil.java:10)
>                                                                                at java.lang.Class.classForName(Native Method)
>                                                                                at java.lang.Class.forName(Class.java:400)
>                                                                                at org.bytedeco.javacpp.Loader.load(Loader.java:585)
>                                                                                at org.bytedeco.javacpp.Loader.load(Loader.java:546)
>                                                                                at org.bytedeco.javacv.FFmpegFrameGrabber.tryLoad(FFmpegFrameGrabber.java:93)
>                                                                                at org.bytedeco.javacv.FFmpegFrameGrabber.<clinit>(FFmpegFrameGrabber.java:118)
>                                                                                at firenation.beepbopP.Feedback.FeedbackActivity.onCreate(FeedbackActivity.java:35)
>                                                                                at android.app.Activity.performCreate(Activity.java:6757)
>                                                                                at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
>                                                                                at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2679)
>                                                                                at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2787)
>                                                                                at android.app.ActivityThread.-wrap12(ActivityThread.java)
>                                                                                at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1504)
>                                                                                at android.os.Handler.dispatchMessage(Handler.java:102)
>                                                                                at android.os.Looper.loop(Looper.java:154)
>                                                                                at android.app.ActivityThread.main(ActivityThread.java:6247)
>                                                                                at java.lang.reflect.Method.invoke(Native Method)
>                                                                                at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:872)
>                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:762)
> 03-11 17:13:16.817 11910-11910/firenation.beepbopP.ember E/AndroidRuntime: Caused by: java.lang.UnsatisfiedLinkError: dalvik.system.PathClassLoader[DexPathList[[zip file "/data/app/firenation.beepbopP.ember-1/base.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_dependencies_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_0_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_1_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_2_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_3_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_4_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_5_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_6_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_7_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_8_apk.apk", zip file "/data/app/firenation.beepbopP.ember-1/split_lib_slice_9_apk.apk"],nativeLibraryDirectories=[/data/app/firenation.beepbopP.ember-1/lib/arm64, /data/app/firenation.beepbopP.ember-1/base.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_dependencies_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_0_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_1_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_2_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_3_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_4_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_5_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_6_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_7_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_8_apk.apk!/lib/arm64-v8a, /data/app/firenation.beepbopP.ember-1/split_lib_slice_9_apk.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64]]] couldn't find "libavutil.so"
>                                                                                at java.lang.Runtime.loadLibrary0(Runtime.java:972)
>                                                                                at java.lang.System.loadLibrary(System.java:1530)
>                                                                                at org.bytedeco.javacpp.Loader.loadLibrary(Loader.java:804)
>                                                                                at org.bytedeco.javacpp.Loader.load(Loader.java:604)
>                                                                                	... 21 more

**Comment by saudet** [comment]
`couldn't find "libjniavutil.so"` -> You'll need to provide that file for your application to work. It looks like you need a 64-bit version of the binaries, so marking as a duplicate of https://github.com/bytedeco/javacpp-presets/issues/52.

**Comment by parejadan** [comment]
@saudet would I still need to rebuild the FFmpeg even if it shows up within the jar?

![capture](https://cloud.githubusercontent.com/assets/11270882/23831789/60de178e-06fe-11e7-96a8-a75b87c634e3.PNG)

I was also following [issue 151](https://github.com/bytedeco/javacpp-presets/issues/151) by adding

**Comment by saudet** [comment]
If you ran your application in 32-bit mode, all would be fine, but you're running it in 64-bit mode for some reason. If you need to run it in 64-bit mode, you'll also need to have 64-bit binaries for FFmpeg.

**Comment by parejadan** [comment]
The device I'm working with is arm64, and I was just about to ask if it could work with 32 but a [quick search](http://stackoverflow.com/questions/30782848/how-to-use-32-bit-native-libaries-on-64-bit-android-device) shed some light as to why it was defaulting to 64.

If you look at my folder structure I might want to just keep the one's I've highlighted (searched for 32 bits)

![capture](https://cloud.githubusercontent.com/assets/11270882/23832065/25d9a3aa-0704-11e7-87fc-6c44ef9be25f.PNG)

The reason I was originally using them was because I read somewhere opencv 3.0 natively only works for 64 bit? (cpp code you see is using it). But since I'm using javacv and it's a wrapper in existing native opencv code, would I get the same performance and support if I just translate that code back to java?

**Comment by saudet** [comment]
OpenCV doesn't need 64-bit, it works fine in 32-bit mode as well.

**Comment by parejadan** [comment]
It works! :DDDDDDDDDDD

I looked at the ffmpeg jar and matched the lib support in my gradel and jniLibs folder (only scoped it to armabi in jniLibs support because gradle setup isn't enough) 

Thanks @saudet !

**Comment by saudet** [comment]
64-bit builds for Android (as the "android-arm64" and "android-x86_64" platforms) are now available, so please give it a try! Binaries for 1.4.1-SNAPSHOT can be downloaded from the snapshot repository: http://bytedeco.org/builds/
